core/extensions/impl/crypto_extension.cpp

```diff
@@ -188,60 +194,91 @@ namespace kagome::extensions {
 
   // ---------------------- runtime api version 1 methods ----------------------
 
+  namespace {
+    auto encode_result = [](auto &&value) -> common::Buffer {
+      static auto logger =
+          common::createLogger("CryptoExtension::scale_encode");
+      auto &&result = scale::encode(value);
+      if (!result) {
+        logger->error("failed to scale-encode value: {} ",
+                      result.error().message());
+        std::terminate();
+      }
+      return common::Buffer(result.value());
+    };
+
+    template <class T>
+    auto encode_optional = [](auto &&v) -> common::Buffer {
+      return encode_result(std::forward<boost::optional<T>>(v));
+    };
+
+    template <class T, class E>
+    auto encode_variant = [](auto &&v) -> common::Buffer {
+      return encode_result(std::forward<boost::variant<T, E>>(v));
+    };
+  }  // namespace
```

Comment:
Remove all that. Scale encode always returns value. So just do ` scale::encode(val).value();` instead